lib/reek/source/source_locator.rb

```diff
@@ -16,22 +18,37 @@ def initialize(paths)
       #
       # Returns a list of Source::SourceCode.
       def sources
-        find_sources.map { |pathname| Source::SourceCode.from File.new(pathname) }
+        source_paths.map { |pathname| Source::SourceCode.from File.new(pathname) }
       end
 
       private
 
-      def find_sources(paths = @paths)
-        paths.map do |path|
-          pathname = Pathname.new(path)
-          if pathname.directory?
-            find_sources(Dir["#{pathname}/**/*.rb"])
-          else
-            next pathname if pathname.file?
-            $stderr.puts "Error: No such file - #{pathname}"
-            nil
+      def source_paths
+        relevant_paths = []
+        @paths.map do |given_path|
+          print_no_such_file_error(given_path) && next unless path_exists?(given_path)
+          Find.find(given_path) do |path|
```

Comment:
>> (Also, are we interested in reek being thread-safe?)

Good question.
So what could be the worst thing that happens?
Somebody using reek via our API in 2 different threads with one thread f***ing up "Find" in the other thread?